kogito-benchmarks-framework/src/main/java/org/kie/kogito/benchmarks/framework/Commands.java

```diff
@@ -0,0 +1,484 @@
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Scanner;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.assertj.core.api.Assertions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.kie.kogito.benchmarks.framework.Logs.appendln;
+import static org.kie.kogito.benchmarks.framework.Logs.appendlnSection;
+
+public class Commands {
+    private static final Logger logger = LoggerFactory.getLogger(Commands.class);
+
+    public static final String BASE_DIR = getBaseDir();
+    public static final String APPS_DIR = getAppsDir();
+    public static final String MVNW = Commands.isThisWindows ? "mvnw.cmd" : "./mvnw";
+    public static final boolean isThisWindows = System.getProperty("os.name").matches(".*[Ww]indows.*");
+    private static final Pattern numPattern = Pattern.compile("[ \t]*[0-9]+[ \t]*");
+    private static final Pattern quarkusVersionPattern = Pattern.compile("[ \t]*<quarkus.version>([^<]*)</quarkus.version>.*");
+    private static final Pattern trailingSlash = Pattern.compile("/+$");
+
+    public static String getArtifactGeneBaseDir() {
+        for (String p : new String[] { "ARTIFACT_GENERATOR_WORKSPACE", "artifact.generator.workspace" }) {
+            String env = System.getenv().get(p);
+            if (StringUtils.isNotBlank(env)) {
+                return env;
+            }
+            String sys = System.getProperty(p);
+            if (StringUtils.isNotBlank(sys)) {
+                return sys;
+            }
+        }
+        return System.getProperty("java.io.tmpdir");
+    }
+
+    public static String getLocalMavenRepoDir() {
+        for (String p : new String[] { "TESTS_MAVEN_REPO_LOCAL", "tests.maven.repo.local" }) {
+            String env = System.getenv().get(p);
+            if (StringUtils.isNotBlank(env)) {
+                return env;
+            }
+            String sys = System.getProperty(p);
+            if (StringUtils.isNotBlank(sys)) {
+                return sys;
+            }
+        }
+
+        String mainBuildRepo = System.getProperty("maven.repo.local");
+        if (StringUtils.isNotBlank(mainBuildRepo)) {
+            return mainBuildRepo;
+        }
+
+        return System.getProperty("user.home") + File.separator + ".m2" + File.separator + "repository";
```

Comment:
wouldn't `Path.resolve()` method chaining be less complicated here?

Reply:
Yes, replaced with `Path.of()`.